xpcPlugin/MessageHandlers.cpp

```diff
@@ -686,21 +689,98 @@ namespace XPC
 			Log::WriteLine(LOG_INFO, "TEXT", "[TEXT] Text set");
 		}
 	}
-
+    
 	void MessageHandlers::HandleView(const Message& msg)
 	{
 		// Update Log
 		Log::FormatLine(LOG_TRACE, "VIEW", "Message Received(Conn %i)", connection.id);
 
+		int enable_camera_location = 0;
+		
 		const std::size_t size = msg.GetSize();
-		if (size != 9)
+		if (size == 9)
 		{
-			Log::FormatLine(LOG_ERROR, "VIEW", "Error: Unexpected length. Message was %d bytes, expected 9.", size);
+			// default view switcher as before
+		}
+		else if (size == 37)
+		{
+			// Allow camera location control
+			enable_camera_location = 1;
+		}
+		else
+		{
+			Log::FormatLine(LOG_ERROR, "VIEW", "Error: Unexpected length. Message was %d bytes, expected 9 or 37.", size);
 			return;
 		}
 		const unsigned char* buffer = msg.GetBuffer();
 		int type = *((int*)(buffer + 5));
 		XPLMCommandKeyStroke(type);
+		
+		if(type == 79 && enable_camera_location == 1) // runway camera view
+		{
+			static struct camera_properties campos;
```

Comment:
Does this need to be `static`?

Reply:
The camera callback accesses this as long at it is being called. So if the variable is destroyed at the end of the function, the camera looses the position information and the screen goes black.

Alternatively this might become a class variable, so it is global in this scope... Or open to any other suggestions.